docs/introduction.md

```diff
@@ -0,0 +1,194 @@
+      "examples": [
+        2
+      ]
+    },
+    "tiffLocation": {
+      "$id": "#/properties/tiffLocation",
+      "type": "string",
+      "title": "The Tifflocation Schema",
+      "default": "",
+      "examples": [
+        "s3://cool-bucket/image.tiff"
+      ],
+      "format": "uri"
+    }
+  }
+}
+```
+
+With this JSON schema attached to our model, it's impossible to create model runs
+with improper or improperly formatted arguments.
+
+Over time, your model's input requirements may change. In that case it makes
+sense to make a new model with a different validator. Since writing JSON schema
+by hand is tedious, it's helpful to use [`json-schema.net`](https://jsonschema.net/)
+to generate schemas from examples, then edit those to match your needs.
+
+# What's a `Prediction`?
+
+A `Prediction` is a single run of a model with specific inputs. Predictions are
+created when you `POST` a model id and arguments to `/api/predictions`.
+
+```scala mdoc
+Prediction.Create(
+  modelId, // the modelId to associate with this prediction
+  ().asJson // the arguments to pass to the model
+).asJson.spaces2
+```
+
+The model's JSON schema is used to validate the prediction's arguments.
+If validation passes, Granary will insert a record for this prediction and submit
+a job to AWS Batch with the resources configured on the model. If that was successful,
+you'll receive a response that looks like this:
+
+```scala mdoc
+Prediction(
+  predictionId,
+  modelId,
+  invocationTime,
+  ().asJson,
+  JobStatus.Started,
+  None,
+  None,
+  Some(webhookId)
+).asJson.spaces2
+```
+
+The `webhookId` in the response points to a single-use webhook for updating the prediction.
+This webhook can be accessed at `/api/predictions/{predictionId}/results/{webhookId}` and
+accepts two kinds of messages:
+
+```scala mdoc
+// JSON of the message to send if the prediction failed
+PredictionFailure("everything went wrong").asJson.spaces2
+
+// JSON of the message to send if the prediction succeeded
+PredictionSuccess("s3://where/the/results/live.json").asJson.spaces2
+```
+
+In the ideal case, the container for running the model in batch will submit results when it
+is done or fails. This strategy will not cover cases in which the model cannot perform
+error-handling though, for instance, `OutOfMemory` errors and cases in which a spot
+instance gets cycled out from under your running model. Because the space of things that
+can go wrong is nearly infinite, Granary itself doesn't provide any facilities for handling
+those sorts of errors or for retrying predictions. Additionally, if your model has retrying
+logic, it's your responsibility to make sure that it doesn't `POST` to the results webhook
+until it has exhausted its retries, since the first `POST` to the webhook will make it
+inaccessible for the rest of time.
+
+# Can I just see some API docs?
+
+Sure! Bring up the [dev environment](./development.md), then hit `localhost:8080/api/docs/docs.yaml`.
```

Comment:
Should we just open up our site (we have it under auth at this point) - https://granary.rasterfoundry.com/api/docs/docs.yaml so that folks don't need to spin up their own server?

Reply:
Uhhh sure that makes sense. I'll include a link.